### PR TITLE
chore: implemented login and endpoint wit tests

### PR DESCRIPTION
**Pull request by sanjam-deriv**

### CODE REVIEW

File: src/pages/auth.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import Layout from '@theme/Layout';
+import { Login } from '../components/Login/Login';
+
+export default function Home(): JSX.Element {
+  return (
+    <Layout title={'Auth'} description='Deriv API documentation'>
```
Comment (matin-deriv) [inline src/pages/auth.tsx:7]:
```suggestion
    <Layout title='Auth' description='Deriv API documentation'>
```

File: src/pages/endpoint.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import Layout from '@theme/Layout';
+import Endpoint from '../components/Endpoint/Endpoint';
+
+export default function Home(): JSX.Element {
+  return (
+    <Layout title={'Endpoint'} description='Deriv API documentation'>
```
Comment (matin-deriv) [inline src/pages/endpoint.tsx:7]:
```suggestion
    <Layout title='Endpoint' description='Deriv API documentation'>
```

File: src/components/Login/tests/Login.test.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  });
+  it('should render title properly', () => {
+    const login_text = screen.getByRole('heading');
+    expect(login_text).toBeInTheDocument();
+  });
+  it('should render image properly', () => {
+    const login_img = screen.getByRole('image');
+    expect(login_img).toHaveStyle('background: /img/login.svg');
+  });
+  // it('should redirect anonymous users to /auth/ when clicking on login button', async () => {
```
Comment (matin-deriv) [inline src/components/Login/tests/Login.test.tsx:25]: let's get rid of the commented code.

File: src/components/Login/Login.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { Button, Text } from '@deriv/ui';
+import { loginUrl } from '../../utils/loginurls';
+import React from 'react';
+import styles from './Login.module.scss';
+
+export const Login = () => {
+  return (
```
Comment (matin-deriv) [inline src/components/Login/Login.tsx:7]:
```suggestion
export const Login = () => (
```

File: src/components/Endpoint/Endpoint.tsx [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+  const default_endpoint = {
+    app_id: '31063',
+    server_url: 'green.binary.com',
+  };
+
+  const [app_id, setAppId] = useState('');
+  const [server_url, setServerUrl] = useState('');
+
+  const handleClick = () => {
+    console.log('Dataset: ', app_id, server_url);
```
Comment (matin-deriv) [inline src/components/Endpoint/Endpoint.tsx:21]: remove `console.log`

File: src/features/Endpoint/Endpoint.module.scss [hunk 6]
```diff
@@ -0,0 +1,104 @@
+.pageContent {
+  height: 140vh;
```
Comment (mohsen-deriv) [inline src/features/Endpoint/Endpoint.module.scss:2]: can we remove this? 
its making the page big and adds a scrollbar, do we need it?

File: src/hooks/useLoginUrl/index.tsx [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import { deriv_urls } from '@site/src/utils/loginurls';
+import { useCallback } from 'react';
+
+const useLoginUrl = (language) => {
+  const getUrl = useCallback(() => {
+    const config_server_url = localStorage.getItem('config.server_url');
```
Comment (mohsen-deriv) [inline src/hooks/useLoginUrl/index.tsx:6]: on first time usage, the `config.server_url` and `config.app_id` is empty and the auth link is broken, can you please add default values for these? 
this is the generated auth link for me `https://oauth.deriv.com/oauth2/authorize?app_id=null&l=en`

File: src/utils/loginurls.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+const domain_url = supported_domains.includes(domain_url_initial)
+  ? domain_url_initial
+  : deriv_api_url;
+
+export const deriv_urls = Object.freeze({
+  DERIV_HOST_NAME: domain_url,
+  DERIV_APP_PRODUCTION: 'https://deriv-api-docs.binary.sx/',
+});
+
+export const loginUrl = ({ language }) => {
```
Comment (mohsen-deriv) [inline src/utils/loginurls.ts:18]: we don't need this anymore right? 
we have this logic as hook, can we remove it?

File: src/utils/loginurls.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
+const isBrowser = () => typeof window !== 'undefined';
+
+const deriv_api_url = 'deriv.com';
```
Comment (mohsen-deriv) [inline src/utils/loginurls.ts:3]: can we move all of these to `utils/constants`?
thanks :pray:

File: src/features/Endpoint/__tests__/Endpoint.test.tsx [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+  afterEach(cleanup);
+
+  it('should render properly', () => {
+    const endpoint = screen.getByTestId('Endpoint');
+    expect(endpoint).toBeInTheDocument();
+  });
+  it('should render title properly', () => {
+    const endpoint_text = screen.getByRole('heading');
+    expect(endpoint_text).toBeInTheDocument();
```
Comment (mohsen-deriv) [inline src/features/Endpoint/__tests__/Endpoint.test.tsx:19]: please check the text content in the heading as well. thanks :pray:

File: src/hooks/useLoginUrl/__tests__/useLoginUrl.test.tsx [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+
+describe('Login', () => {
+  it('Should return a string containing oauth2', () => {
+    const { result } = renderHook(() => useLoginUrl('en'));
+    expect(result.current.getUrl()).toContain('/oauth2/');
+  });
+  it('Should return a string containing language "en"', () => {
+    const { result } = renderHook(() => useLoginUrl('en'));
+    expect(result.current.getUrl()).toContain('/oauth2/');
+  });
```
Comment (mohsen-deriv) [inline src/hooks/useLoginUrl/__tests__/useLoginUrl.test.tsx:12]: please add test cases to check the `app_id`, `login_url` as well.

File: docusaurus.config.js [hunk 12]
```diff
@@ -64,6 +64,11 @@ const config = {
             label: 'Documentation',
           },
           { to: 'https://tech.deriv.com/', label: 'Blog', position: 'left' },
+          {
+            to: '/auth',
+            position: 'left',
```
Comment (mohsen-deriv) [inline docusaurus.config.js:69]: can we change this to `right`? 
it should look better I guess, what do you think?

File: src/features/Endpoint/__tests__/Endpoint.test.tsx [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+  });
+
+  it('should have default values in input fields', () => {
+    const server = screen.getByPlaceholderText('e.g. frontend.binaryws.com');
+    expect(server).toHaveValue('green.binary.com');
+
+    const app_id = screen.getByPlaceholderText('e.g. 9999');
+
+    expect(app_id).toHaveValue('35014');
+  });
```
Comment (mohsen-deriv) [inline src/features/Endpoint/__tests__/Endpoint.test.tsx:29]: please write test cases for point of failures, such as wrong inputs ( form validation, etc ), thanks :pray:

**Review by mohsen-deriv — CHANGES_REQUESTED** [review]

File: src/hooks/useLoginUrl/index.tsx (hunk 7 above)
Comment (sanjam-deriv) [reply src/hooks/useLoginUrl/index.tsx]: Fixed the initial auth

File: src/utils/loginurls.ts (hunk 8 above)
Comment (sanjam-deriv) [reply src/utils/loginurls.ts]: Made the hook

File: src/utils/loginurls.ts (hunk 9 above)
Comment (sanjam-deriv) [reply src/utils/loginurls.ts]: Done

File: src/utils/constants.ts [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+  app_id: '',
+  server_url: '',
+};
+
+export const domain_app_ids = {
+  'app.deriv.com': 16929,
+  'staging-app.deriv.com': 16303,
+};
+
+export const getCurrentProductionDomain = () =>
```
[the remark this answers is not in the document]
Comment (sanjam-deriv) [reply src/utils/constants.ts]: Done

File: src/hooks/useLoginUrl/__tests__/useLoginUrl.test.tsx (hunk 11 above)
Comment (sanjam-deriv) [reply src/hooks/useLoginUrl/__tests__/useLoginUrl.test.tsx:12]: Done

File: docusaurus.config.js (hunk 12 above)
Comment (sanjam-deriv) [reply docusaurus.config.js]: Done

File: src/features/Endpoint/__tests__/Endpoint.test.tsx (hunk 13 above)
Comment (sanjam-deriv) [reply src/features/Endpoint/__tests__/Endpoint.test.tsx:37]: Added

**Review by mohsen-deriv — APPROVED** [review]

**Pull request merged**